Enabled Pairs & Ignore Outliers

HojouFotytu opened the pull request:
- minor resize of dashboard bottom charts

- fixed an error where underscore was added to Binance Futures coin pairs (issue #230)

HojouFotytu commented:
Fix for issue #233.

An optional field can be added to MarketTrends in the analyzer:

"IgnoreOutlier": X
Where X is an integer representing a limit on a pair's inclusion in generating the market trend.

**Example:**
       {
          "Name": "12h",
          "Platform": "Exchange",
          "MaxMarkets": 75,
          "TrendMinutes": 1440,
	  "TrendCurrency": "Market",
	  "IgnoreOutlier": 50,
	  "DisplayGraph": true
        },

Any coin pair with a 12h trend that is above 50% or below -50% will not be used when calculating the global 12h trend.

djbadders commented on Core/MarketAnalyzer/BaseAnalyzer.cs:
```diff
@@ -429,7 +429,29 @@ public static void SaveMarketInfosToFile(Dictionary<string, MarketInfo> marketIn
+                }
+              }
+
+              double averageTrendChange = totalTrendChange / marketTrendChanges.Count;
```
This average calculation is wrong as you should only be dividing against the number of entries that contributed to the total. You need to maintain a count alongside adding to the total and divide by this to find the average.

HojouFotytu replied:
Good catch.  Thanks.

HojouFotytu replied:
This should fix it:  https://github.com/PTMagicians/PTMagic/pull/231/commits/65bb1dc719c6472c0a0033f0b0b2d32d494d3a95

djbadders commented on Core/DataObjects/PTMagicData.cs:
```diff
@@ -146,6 +146,9 @@ public class MarketTrend
     public string AllowedMarkets { get; set; } = "";
 
+    [DefaultValue(0)]
+    public int IgnoreOutlier { get; set; } = 0;
```
Can I suggest a rename as IgnoreOutlier sounds like a boolean value, when it is actually a threshold value. I suggest IgnoreOutlierThreshold

HojouFotytu replied:
How about TrendThreshold?  Would match some of the other MarketTrend nomenclature...

HojouFotytu commented on Core/MarketAnalyzer/BaseAnalyzer.cs:
```diff
@@ -429,7 +429,29 @@ public static void SaveMarketInfosToFile(Dictionary<string, MarketInfo> marketIn
+                {
+                  if ((marketTrendChange.TrendChange > marketTrend.IgnoreOutlier) || (marketTrendChange.TrendChange < (marketTrend.IgnoreOutlier * -1)))
+                  {
+                    log.DoLogWarn("Market trend '" + marketTrend.Name + "' is ignoring the outlier '" + marketTrendChange.Market + ".");
```
Absolutely.  I set it as warning so it was easy for me to find during testing.

HojouFotytu commented:
Changed "IgnoreOutlier" to "TrendThreshold"

HojouFotytu commented:
Added Threshold % column to Market Trends on DashboardBottom and MarketAnalyzer

Added a fix to prevent the display of DYNamic formulas in the pairs list.

HojouFotytu commented:
Added last runtime to ticker health-icon tooltip.

Fixed a typo for the URL in the current global setting info-icon on the ticker.

HojouFotytu commented:
Fix for issue #229.

The addition of the Profit Target column in the Pairs list with version 2.4.4 was causing an issue for pairs that had no profit target.

HojouFotytu commented:
fix for column alignment for coins without a profit target.

djbadders reviewed (approved):
Looks good, nice work
